src/test/java/teammates/sqlui/webapi/CalculateUsageStatisticsActionTest.java

```diff
@@ -0,0 +1,137 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.UsageStatisticsAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.storage.sqlentity.UsageStatistics;
+import teammates.ui.webapi.CalculateUsageStatisticsAction;
+
+/**
+ * SUT: {@link CalculateUsageStatisticsAction}.
+ */
+public class CalculateUsageStatisticsActionTest extends BaseActionTest<CalculateUsageStatisticsAction> {
+    private static final int NUMBER_OF_RESPONSES = 2;
+    private static final int NUMBER_OF_COURSES = 2;
+    private static final int NUMBER_OF_STUDENTS = 2;
+    private static final int NUMBER_OF_INSTRUCTORS = 2;
+    private static final int NUMBER_OF_ACCOUNT_REQUESTS = 2;
+    private static final int COLLECTION_TIME_PERIOD = 60;
+    Instant endTime = TimeHelper.getInstantNearestHourBefore(Instant.now());
+    Instant startTime = endTime.minus(COLLECTION_TIME_PERIOD, ChronoUnit.MINUTES);
+    UsageStatistics testUsageStatistics;
+    UsageStatisticsAttributes testUsageStatisticsAttributes;
+
+    @Override
+    protected String getActionUri() {
+        return Const.CronJobURIs.AUTOMATED_USAGE_STATISTICS_COLLECTION;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return GET;
+    }
+
+    @Test
+    void testAccessControl_admin_canAccess() {
+        verifyCanAccess();
+    }
+
+    @Test
+    void testAccessControl_maintainers_cannotAccess() {
+        logoutUser();
+        loginAsMaintainer();
+        verifyCannotAccess();
+    }
+
+    @Test
+    void testAccessControl_instructor_cannotAccess() {
+        logoutUser();
+        loginAsInstructor(Const.ParamsNames.INSTRUCTOR_ID);
+        verifyCannotAccess();
+    }
+
+    @Test
+    void testAccessControl_student_cannotAccess() {
+        logoutUser();
+        loginAsStudent(Const.ParamsNames.STUDENT_ID);
+        verifyCannotAccess();
+    }
+
+    @Test
+    void testAccessControl_loggedOut_cannotAccess() {
+        logoutUser();
+        verifyCannotAccess();
+    }
+
+    @Test
+    void testAccessControl_unregistered_cannotAccess() {
+        logoutUser();
+        loginAsUnregistered(Const.ParamsNames.USER_ID);
+        verifyCannotAccess();
+    }
+
+    @BeforeMethod
+    void setUp() {
+        loginAsAdmin();
+        testUsageStatistics = new UsageStatistics(
+                startTime,
+                COLLECTION_TIME_PERIOD,
+                NUMBER_OF_RESPONSES,
+                NUMBER_OF_COURSES,
+                NUMBER_OF_STUDENTS,
+                NUMBER_OF_INSTRUCTORS,
+                NUMBER_OF_ACCOUNT_REQUESTS,
+                0,
+                0);
```

Comment:
I think we can create a `getTypicalUsageStatistics()` method in `BaseTestCase.java` with all of these values/attributes over there for extensibility.